test/test_stream.jl

```diff
@@ -12,14 +12,10 @@ buf = IOBuffer()
 pipe = Base64EncodePipe(buf)
 write_to_png(c,pipe)
 close(pipe)
+
 # Catch short writes
 
-    if VERSION >= v"0.6.0-dev.1954"
-        str = String(take!(buf))
-        str_data = Vector{UInt8}(str)
-    else
-        str = takebuf_string(buf)    
-        str_data = str.data
-    end        
+@compat str = String(take!(buf))
```

Comment:
It's not entirely unique to Compat, it's whether or not something is valid syntax on older versions of Julia. If it is, there's no need for a macro.